zipline/utils/preprocess.py

```diff
@@ -84,26 +84,19 @@ def _decorator(f):
         if defaults is None:
             defaults = ()
         no_defaults = (NO_DEFAULT,) * (len(args) - len(defaults))
-        args_defaults = zip(args, no_defaults + defaults)
-
-        argset = set(args)
-
-        # These assumptions simplify the implementation significantly.  If you
-        # really want to validate a *args/**kwargs function, you'll have to
-        # implement this here or do it yourself.
+        args_defaults = list(zip(args, no_defaults + defaults))
         if varargs:
-            raise TypeError(
-                "Can't validate functions that take *args: %s" % argspec
-            )
+            args_defaults.append((varargs, NO_DEFAULT))
         if varkw:
-            raise TypeError(
-                "Can't validate functions that take **kwargs: %s" % argspec
-            )
+            args_defaults.append((varkw, NO_DEFAULT))
+
+        argset = set(args) | {varargs, varkw} - {None}
```

Comment:
Can we put parens on the RHS here?  I don't have immediate intuition for whether `|` or `-` binds tighter.